jetty-home/src/main/resources/bin/jetty.sh

```diff
@@ -435,7 +433,7 @@ CYGWIN*) JETTY_START="`cygpath -w $JETTY_START`";;
 esac
 
 RUN_ARGS=$("$JAVA" -jar "$JETTY_START" --dry-run=opts,path,main,args ${JETTY_ARGS[*]} ${JAVA_OPTIONS[*]})
-RUN_CMD=("$JAVA" $JETTY_SYS_PROPS ${RUN_ARGS[@]})
+RUN_CMD=($JETTY_SYS_PROPS ${RUN_ARGS[@]})
```

Comment:
The `RUN_ARGS` line is executing `java -jar start.jar --dry-run=opts,path,main,args` and capturing the output in `RUN_ARGS`
The `RUN_CMD` doesn't have `$JAVA` as that's one of the problems causing the correct execution of the jetty.sh with awkward jetty properties.
The later command that takes `$RUN_CMD` and passes it through `xargs` is splitting the args up properly and then passing that into `$JAVA` correctly.

Reply:
I think we should swap the names around. Currently `RUN_ARGS` does contain the java command (as it is output by -dry-run), but `RUN_CMD` does not.   It is cooking my noodle and making it hard to understand the rest of the code.

Reply:
Please note, if your remove `$JAVA` from `$RUN_CMD`, you need to update a code for `run` action.

Reply:
`RUN_ARGS` does not contain the `java` portion, simply because of the limitation on the `--dry-run` selection.

Example:

``` shell
[jetty-home/target/foobase][fix/jetty-10.0.x/jetty-sh-start-properties]$ java -jar ../jetty-home/start.jar --dry-run=opts,path,main,args
-Djava.io.tmpdir=/tmp -Djetty.home=/home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home -Djetty.base=/home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/foobase --class-path /home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/foobase/resources:/home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/lib/logging/slf4j-api-2.0.5.jar:/home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/lib/logging/jetty-slf4j-impl-10.0.14-SNAPSHOT.jar:/home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/lib/jetty-servlet-api-4.0.6.jar:/home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/lib/jetty-http-10.0.14-SNAPSHOT.jar:/home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/lib/jetty-server-10.0.14-SNAPSHOT.jar:/home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/lib/jetty-xml-10.0.14-SNAPSHOT.jar:/home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/lib/jetty-util-10.0.14-SNAPSHOT.jar:/home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/lib/jetty-io-10.0.14-SNAPSHOT.jar:/home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/lib/jetty-security-10.0.14-SNAPSHOT.jar:/home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/lib/jetty-servlet-10.0.14-SNAPSHOT.jar:/home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/lib/jetty-webapp-10.0.14-SNAPSHOT.jar:/home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/lib/jetty-deploy-10.0.14-SNAPSHOT.jar org.eclipse.jetty.xml.XmlConfiguration 'java.version'='17.0.6' 'java.version.major'='17' 'java.version.micro'='6' 'java.version.minor'='0' 'java.version.platform'='17' 'jetty.base'='/home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/foobase' 'jetty.base.uri'='file:///home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/foobase' 'jetty.home'='/home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home' 'jetty.home.uri'='file:///home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home' 'jetty.webapp.addServerClasses'='org.eclipse.jetty.logging.,file:///home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/lib/logging/,org.slf4j.' 'runtime.feature.alpn'='true' 'slf4j.version'='2.0.5' /home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/etc/jetty-bytebufferpool.xml /home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/etc/jetty-threadpool.xml /home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/etc/jetty.xml /home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/etc/jetty-webapp.xml /home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/etc/jetty-deploy.xml /home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/etc/jetty-http.xml /home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/etc/jetty-requestlog.xml
```

This is because the dry-run part `java` (aka `--dry-run=java`) is not part of that command.

Reply:
Please look at the line 598.